src/test/java/jep/test/TestSharedInterpreter.java

```diff
@@ -86,6 +92,53 @@ public static void testSharedTypes() throws Throwable{
         }
     }
 
+    public static void testSharedJPyObject() throws Throwable {
+        try (Interpreter interp = new SharedInterpreter()) {
+            PyObject list = interp.getValue("[]", PyObject.class);
+            Thread t = new Thread(() -> {
+                try (Interpreter interp2 = new SharedInterpreter()) {
+                    list.getAttr("append", PyCallable.class).call("test");
+                    interp2.set("l", list);
+                    interp2.exec("l.append('test2')");
+                }
+            });
+            t.start();
+            t.join();
+            String test2 = list.getAttr("pop", PyCallable.class).callAs(String.class);
+            String test = list.getAttr("pop", PyCallable.class).callAs(String.class);
+            if (!test.equals("test")) {
```

Comment:
Best practice is to put the constant string on the left side of the .equals() call.  `!"test".equals(test)`

Reply:
Fixed